include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp

```diff
@@ -634,5 +635,272 @@ struct BlockToCTileMap_3DGrid_KSplit
         return true;
     }
 };
+#include <stdlib.h>
+template <uint32_t MPerBlock_, uint32_t NPerBlock_, uint32_t KPerBlock_>
+struct BlockToCTileMap_GemmStreamK
+{
+    static constexpr uint32_t min_k_iters_per_sk_block = 2;
+    static constexpr uint32_t MPerBlock                = MPerBlock_;
+    static constexpr uint32_t NPerBlock                = NPerBlock_;
+    static constexpr uint32_t KPerBlock                = KPerBlock_;
+
+    //--------------------------------------
+    // pass to device
+    uint32_t sk_num_blocks;
+    uint32_t sk_num_big_blocks;
+    uint32_t sk_total_iters;
+
+    uint32_t dp_start_block_idx;
+    uint32_t dp_iters_per_block;
+    uint32_t dp_num_blocks;
+
+    uint32_t k_iters_per_big_block;
+    MDiv k_iters_per_tile;
+    MDiv n_tiles;
+    MDiv tile_swizzle_sub_m;
+    MDiv tile_swizzle_sub_m_rem;
+    //--------------------------------------
+
+    static int env_get_int(const char* var_name, int default_int)
+    {
+        char* v = getenv(var_name);
+        int r   = default_int;
+        if(v)
+            r = atoi(v);
+        return r;
+    }
+
+    // prefer construct on host
+    BlockToCTileMap_GemmStreamK(
+        uint32_t m, uint32_t n, uint32_t k, uint32_t num_cu, uint32_t occupancy, uint32_t tile_swizzle_sub_m_factor = 8)
+    {
+        uint32_t num_tiles =
+            math::integer_divide_ceil(m, MPerBlock) * math::integer_divide_ceil(n, NPerBlock);
+        k_iters_per_tile = MDiv(math::integer_divide_ceil(k, KPerBlock));
+
+        // one cu can hold one wg at one time, from the whole chip's point of view
+        // if number of wg is same as num_cu, we call it 1 dispatch
+        // if number of wg is 2x num_cu, we call it 2 dispatches.
+        // one dispatch can deliever wg same as num_cu (full dispatch), or less than num_cu (partial
```

Comment:
```suggestion
        // one dispatch can deliver wg same as num_cu (full dispatch), or less than num_cu (partial
```

Reply:
@keryell thanks for this review, I'll fix this problem in later commit